packages/icon-button/src/IconButton.tsx

```diff
@@ -200,11 +201,13 @@ const IconButton: ExtendableBox<
     const mode = darkMode ? 'dark' : 'light';
 
     // We do our own proptype validation here to ensure either 'aria-label' or 'aria-labelledby' are passed to the component.
-    if (!ariaLabel && !ariaLabelledBy) {
-      console.error(
-        'For screen-reader accessibility, aria-label or aria-labelledby must be provided to IconButton.',
-      );
-    }
+    validateAriaLabelProps(
+      {
```

Comment:
The function was designed to accept all of props. Is there a reason we're doing this manually? It's not actually validating the props passed in to the component.

Reply:
Ditto for all other instances where this approach was taken.

Reply:
They were already pulled out from rest so I just went with it, but I can go back and clean it up where relevant.